torch_geometric/data/remote_backend.py

```diff
@@ -0,0 +1,72 @@
+# This file defines a set of utilities for remote backends (backends that are
+# characterize as Tuple[FeatureStore, GraphStore]).
+from typing import Tuple, Union
+
+from torch_geometric.data.feature_store import FeatureStore
+from torch_geometric.data.graph_store import GraphStore
+from torch_geometric.typing import EdgeType, NodeType
+
+
+def num_nodes(
```

Comment:
I actually like the previous implementation for each feature store and graph store better. That is more intuitive to me compare with get all stores and query the nodes with a query (which sounds more heavy weight).
Could we just expose num_nodes and sizes for features and edges ?

Reply:
Sure, we can expose num nodes for features and edges, but there are instances where an edge index is not part of a graph (e.g. link prediction) or we just want to get the number of nodes from a group name, not a full edge index type. I also don't think it's too heavyweight to query all the edge attributes and node attributes, since these are small data structures and are somewhat bounded. 

However, if you think this overhead would be significant, we can expose methods in the feature store and graph store to get a TensorAttr/EdgeAttr from a partial specification, and directly query these (instead of listing all of them and iterating). This can be done as an improvement in a separate PR. Wdyt?

Reply:
After a second look, I think the diverge comes from whether we want to introduce this new `remote_backend` or consolidate logic in `GraphStore`, i.e we can consider `GraphStore` to be a backend that can be `remote`. And then have an api like
```
class GraphStore:
  def num_nodes(Union[NodeType, EdgeType]):
```
We can also hide `FeatureStore` behind `GraphStore` or `FeatureStore` metadata as part of information in `GraphStore`.

If we want to introduce `remote_backend`, you may want to think more about whether it is a `Backend` object or it is a helper provide util functions talk to `FeatureStore` and `GraphStore`.

Reply:
A few clarifications
* Yes, a GraphStore can be remote.
* We do not want to hide FeatureStore behind GraphStore; ideally, a feature store owns features, and a graph store owns the graph. The complications come because there can be edges in the graph store with no corresponding features in the feature store, etc. as mentioned above.
* `remote_backend` is not a `Backend` object; it's really a helper providing utility functions to talk to `FeatureStore` and `GraphStore` (as in the comment at the top of `remote_backend.py`). However, I agree that the name is confusing. I can change it to `remote_backend_util.py`.